spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigMapPropertySource.java

```diff
@@ -41,25 +44,21 @@
 	private static final Log LOG = LogFactory.getLog(Fabric8ConfigMapPropertySource.class);
 
 	public Fabric8ConfigMapPropertySource(KubernetesClient client, String name) {
-		this(client, name, null, (Environment) null);
+		this(client, name, null, null);
 	}
 
-	public Fabric8ConfigMapPropertySource(KubernetesClient client, String name, String namespace,
+	public Fabric8ConfigMapPropertySource(KubernetesClient client, String applicationName, String namespace,
 			Environment environment) {
-		super(getName(name, getNamespace(client, namespace)),
-				asObjectMap(getData(client, name, getNamespace(client, namespace), environment)));
-	}
-
-	private static String getNamespace(KubernetesClient client, String namespace) {
-		return StringUtils.isEmpty(namespace) ? client.getNamespace() : namespace;
+		super(getName(applicationName, getNamespace(client, namespace)),
+				getData(client, applicationName, getNamespace(client, namespace), environment));
```

Comment:
careful here, as this removes one call : `asObjectMap`. If you look what `asObjectMap` is trying to achieve, it is simply converting a `Map` to a `LinkedHashMap`. Since such a change is not needed (surely looks like a initial artifact of an implementation), this call can be elided away.